--- ../fox-1.7.25.orig/lib/FXDir.cpp	2010-09-17 20:59:29.000000000 -0500
+++ ./lib/FXDir.cpp	2010-12-02 03:08:53.000000000 -0600
@@ -172,12 +172,16 @@
 FXbool FXDir::create(const FXString& path,FXuint perm){
   if(!path.empty()){
 #ifdef WIN32
-#ifdef UNICODE
+# ifdef __MINGW32__
+    return ::mkdir(path.text())==0;
+# else
+#  ifdef UNICODE
     FXnchar buffer[MAXPATHLEN];
     utf2ncs(buffer,MAXPATHLEN,path.text(),path.length()+1);
     return CreateDirectoryW(buffer,NULL)!=0;
-#else
+#  else
     return CreateDirectoryA(path.text(),NULL)!=0;
+# endif
 #endif
 #else
     return ::mkdir(path.text(),perm)==0;
