--- ../fox-1.7.29.orig/lib/FXFile.cpp	2011-07-13 13:34:57.000000000 -0500
+++ ./lib/FXFile.cpp	2011-08-09 22:34:24.000000000 -0500
@@ -402,15 +402,20 @@
 FXbool FXFile::rename(const FXString& srcfile,const FXString& dstfile){
   if(srcfile!=dstfile){
 #ifdef WIN32
-#ifdef UNICODE
+# ifdef __MINGW32__
+    ::unlink(dstfile.text());
+    return ::rename(srcfile.text(),dstfile.text())==0;
+# else
+#  ifdef UNICODE
     FXnchar srcname[MAXPATHLEN];
     FXnchar dstname[MAXPATHLEN];
     utf2ncs(srcname,srcfile.text(),MAXPATHLEN);
     utf2ncs(dstname,dstfile.text(),MAXPATHLEN);
     return ::MoveFileExW(srcname,dstname,MOVEFILE_REPLACE_EXISTING)!=0;
-#else
+#  else
     return ::MoveFileExA(srcfile.text(),dstfile.text(),MOVEFILE_REPLACE_EXISTING)!=0;
-#endif
+#  endif
+# endif
 #else
     return ::rename(srcfile.text(),dstfile.text())==0;
 #endif
