--- ../fox-1.7.27.orig/lib/FXFile.cpp	2011-06-24 23:43:28.000000000 -0500
+++ ./lib/FXFile.cpp	2011-06-29 23:08:25.000000000 -0500
@@ -390,15 +390,20 @@
 FXbool FXFile::rename(const FXString& srcfile,const FXString& dstfile){
   if(srcfile!=dstfile){
 #ifdef WIN32
-#ifdef UNICODE
+# ifdef __MINGW32__
+    ::unlink(dstfile.text());
+    return ::rename(srcfile.text(),dstfile.text())==0;
+# else
+#  ifdef UNICODE
     FXnchar srcname[MAXPATHLEN];
     FXnchar dstname[MAXPATHLEN];
     utf2ncs(srcname,srcfile.text(),MAXPATHLEN);
     utf2ncs(dstname,dstfile.text(),MAXPATHLEN);
     return ::MoveFileExW(srcname,dstname,MOVEFILE_REPLACE_EXISTING)!=0;
-#else
+#  else
     return ::MoveFileExA(srcfile.text(),dstfile.text(),MOVEFILE_REPLACE_EXISTING)!=0;
-#endif
+#  endif
+# endif
 #else
     return ::rename(srcfile.text(),dstfile.text())==0;
 #endif
