--- ../fox-1.7.31.orig/lib/FXFileList.cpp	2011-10-18 17:59:03.000000000 -0500
+++ ./lib/FXFileList.cpp	2011-10-20 07:41:28.000000000 -0500
@@ -1325,6 +1325,11 @@
   FXStat       info;
   FXDir        dir;
 
+#ifdef WIN32
+  if ((directory.length()==3) && (directory[1]==':') && (directory[2]=='\\')) {
+    directory.trunc(2);
+    }
+#endif
   // Get directory stream pointer
   if(dir.open(directory)){
 
