--- ../fox-1.7.25.orig/lib/FXStat.cpp	2010-09-01 21:34:53.000000000 -0500
+++ ./lib/FXStat.cpp	2010-12-02 04:31:47.000000000 -0600
@@ -202,7 +202,7 @@
   info.fileIndex=0;
   info.fileSize=0;
   if(!file.empty()){
-#ifdef WIN32
+#if defined(WIN32) && !defined(__MINGW32__)
 #ifdef UNICODE
     TCHAR buffer[MAXPATHLEN];
     HANDLE hfile;
@@ -261,18 +261,36 @@
 #else
     const FXTime seconds=1000000000;
     struct stat data;
+#ifdef WIN32
+    if (strchr("/\\", file[file.length()-1])) {
+      FXString tmpname=file;
+      while ((!tmpname.empty()) && strchr("/\\", tmpname[tmpname.length()-1])) {
+        tmpname.trunc(tmpname.length()-1);
+      }
+      if ((tmpname.length()==2) && Ascii::isLetter(tmpname[0]) && (tmpname[1]==':')) {
+        tmpname+=PATHSEPSTRING;
+      }
+      if(tmpname.empty() || (::stat(tmpname.text(),&data)!=0)) { return false; }
+    } else {
+      if (::stat(file.text(),&data)!=0) { return false; }
+    }
+    {
+#else
     if(::stat(file.text(),&data)==0){
+#endif
       info.modeFlags=(data.st_mode&0777);
       if(S_ISDIR(data.st_mode)) info.modeFlags|=FXIO::Directory;
       if(S_ISREG(data.st_mode)) info.modeFlags|=FXIO::File;
-      if(S_ISLNK(data.st_mode)) info.modeFlags|=FXIO::SymLink;
       if(S_ISCHR(data.st_mode)) info.modeFlags|=FXIO::Character;
       if(S_ISBLK(data.st_mode)) info.modeFlags|=FXIO::Block;
       if(S_ISFIFO(data.st_mode)) info.modeFlags|=FXIO::Fifo;
+#ifndef WIN32
+      if(S_ISLNK(data.st_mode)) info.modeFlags|=FXIO::SymLink;
       if(S_ISSOCK(data.st_mode)) info.modeFlags|=FXIO::Socket;
       if(data.st_mode&S_ISUID) info.modeFlags|=FXIO::SetUser;
       if(data.st_mode&S_ISGID) info.modeFlags|=FXIO::SetGroup;
       if(data.st_mode&S_ISVTX) info.modeFlags|=FXIO::Sticky;
+#endif
       info.userNumber=data.st_uid;
       info.groupNumber=data.st_gid;
       info.linkCount=data.st_nlink;
@@ -479,7 +497,7 @@
 // Return true if file exists
 FXbool FXStat::exists(const FXString& file){
   if(!file.empty()){
-#ifdef WIN32
+#if defined(WIN32) && !defined(__MINGW32__)
 #ifdef UNICODE
     FXnchar name[MAXPATHLEN];
     utf2ncs(name,MAXPATHLEN,file.text(),file.length()+1);
